### PR TITLE
Nikos/5835/websocket provider keeps important error message back

### LINKED ISSUE #5835: WebsocketProvider keeps important error messages back.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It would be great if we would get the errors with their description from the underlying ``websocket`` package shown to us. 

### Expected Behavior

I do get the underlying errors displayed.

### Steps to Reproduce

Try to connect to a WS endpoint where the upgrade is failing. Basically, all error cases where [this](https://github.com/theturtle32/WebSocket-Node/blob/master/lib/WebSocketClient.js#L335) method is used would provide helpful details. Same for [this case](https://github.com/theturtle32/WebSocket-Node/blob/master/lib/WebSocketClient.js#L225).

### Web3.js Version

<=1.8.x

### Environment

- Operating System: 
- Browser: 
- Node.js Version: 
- NPM Version: 

### Anything Else?

_No response_

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #5835 

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-core-helpers/src/errors.js [hunk 1]
```diff
@@ -73,6 +73,10 @@ module.exports = {
         if (event) {
             error.code = event.code;
             error.reason = event.reason;
+            console.log("the event description", event.description)
```
Comment (jdevcs) [inline packages/web3-core-helpers/src/errors.js]: console.log should be removed

**Review by jdevcs — COMMENTED** [review]
also changelog update is required

**Comment by miraclezrus** [comment]
Pardon?

**Comment by miraclezrus** [comment]
Ok new at this I'll update it later

File: packages/web3-providers-ws/src/index.js [hunk 2]
```diff
@@ -207,7 +247,11 @@ WebsocketProvider.prototype._addSocketListeners = function () {
     this.connection.addEventListener('message', this._onMessage.bind(this));
     this.connection.addEventListener('open', this._onConnect.bind(this));
     this.connection.addEventListener('close', this._onClose.bind(this));
-};
+    if(this.connection._connection){
+        this.connection._client.removeAllListeners('connectFailed'); //Override the internal listeners, so they don't trigger a `close` event. We want to trigger `_onClose` manually with a description.
```
Comment (jdevcs) [inline packages/web3-providers-ws/src/index.js:251]: As browsers use the native [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) object + doesn't use/have ws lib client and node uses ws client. so can we more specifically check `_client` here?

**Review by jdevcs — APPROVED** [review]

**Review by avkos — APPROVED** [review]
LGTM

File: packages/web3-providers-ws/src/index.js (hunk 2 above)
Comment (Muhammad-Altabba) [reply packages/web3-providers-ws/src/index.js:251]: And do we also need to handle the same for the browser WebSocket? Or is it only an issue that we need to handle just for node?

**Pull request merged**